### ISSUE TITLE
Auto decrypt mail

### ISSUE THREAD

**Issue by celogeek** [the-djmaze/snappymail#1142]
**Is your feature request related to a problem? Please describe.**
When I open a crypted message, I need to click on Decrypt to see it.
For example on ProtonMail, the message is decrypted, and I have an icon to tell me that's a crypted and signed message.

**Describe the solution you'd like**
Can we have an option to automatically decrypt encrypted mail ?

**Comment by the-djmaze** [comment]
SnappyMail didn't store decrypt passwords, but the latest version should with the "remember option".
When you use that, you don't fill in your password each time.

However, since we forget to lock our computer when we walk away.
Someone can now decrypt your instantly and read it.
Since the security is gone now, your request could be implemented when password is remembered.

**Comment by celogeek** [comment]
Indeed, if the password is remember, it will save a click on each message.
- You see it once
- click on decrypt
- fill your password with remember
- then each other encrypted message are automatically decrypted

Also what I observe on ProtonMail for example, is that the gpg key used for encryption looks like to be the one for connecting to your account.
So if you can log in, then as all email are encrypted with your password, you can access to them.
They are all encrypted.

**Comment by celogeek** [comment]
<img width="671" alt="Capture d’écran 2023-05-30 à 15 21 50" src="https://github.com/the-djmaze/snappymail/assets/65178/ef0b0b04-e92e-45f4-b86d-1cf4369932fd">

Here:

Graphic: (use Imagick: best one)
- Imagick | ✔
- Gmagick | ❌
- GD | ✔

**Comment by celogeek** [comment]
May be a description of the usage you have with the dependencies, so the people will install it if it's needed for them.

**Comment by randolf** [comment]
Perhaps this behavour should be an option?  If so, it could be included in the Security Settings as a simple set of radio buttons (with the default being the current behaviour of requiring the user to click on "Decrypt" first since this should be regarded as a more secure practice):

`Automatic decryption in message reader:  (_) Yes / (X) No (recommended)`

**Comment by celogeek** [comment]
The PGP key is store in memory after you seize it once.
So I'm not sure automatic decryption is less secure as anyone can just click on decrypt.

It may be useful to loose it after a delay of inactivity instead. So you have to reenter it.

**Comment by the-djmaze** [comment]
> It may be useful to loose it after a delay of inactivity

I've added a delay of 15 minutes.

**Comment by codiflow** [comment]
I would like to invite you also discussing the hardcoded value of 15 minutes here: #1545

I totally get that security should be taken seriously here but having a non-customizable value of 15 minutes leads to other issues like having to enter the PGP password almost every time you want to send and **sign an email** (which most likely happens more often than decrypting mails).

**Comment by mr-gosh** [comment]
I would also like to have an option to stored decrypted messages in the mailbox (for access with software that can't use S/MIME like an Mail Parser for automatic Ticket-Creation.

It is absolutely clear that this can be a security issue - but in our Case we want to receive s/mime encrypted Mails that can than be accessed decrypted by IMAP.

Is this possible or should I open an own issue for that?
